pkg/loki/modules.go

```diff
@@ -395,7 +395,7 @@ func (t *Loki) initQuerier() (services.Service, error) {
 	toMerge := []middleware.Interface{
 		httpreq.ExtractQueryMetricsMiddleware(),
 	}
-	if t.supportIndexDeleteRequest() {
+	if t.supportIndexDeleteRequest() && t.Cfg.CompactorConfig.RetentionEnabled {
```

Comment:
what's this additional check for, why wasn't it previously needed? You check it in combination with `t.supportIndexDeleteRequests` in multiple places so I'm curious if it should be moved into the `t.supportIndexDeleteRequests` function?

Reply:
As mentioned above in the description this extra check is needed when the deletes-based retention is entirely disabled. This is the case by default IIRC and produces red-herring errors in the querier and ruler logs like this:
```
level=error ts=2022-10-18T14:13:41.598649348Z caller=delete_requests_client.go:211 msg="error getting delete requests from the store" err="unexpected status code: 404"
ts=2022-10-18T14:13:41.598697295Z caller=spanlogger.go:80 user=application level=error msg="failed loading deletes for user" err="unexpected status code: 404"
```

I found moving it into the `supportIndexDeleteRequests` obscures the fact that this method is dedicated to check if the underlying index store supports deletes. Adding `RetentionEnabled` in there would make the check for retention hidden/implicit with the store capabilities. OTH as proposed in the implementation it is more explicit for the reader to know that we need an index store that supports deletes and explicitly opt-in to use the feature.